src/org/rascalmpl/parser/gtd/SGTDBF.java

```diff
@@ -1346,82 +1347,85 @@ protected AbstractNode parse(AbstractStackNode<P> startNode, URI inputURI, int[]
 
 	    initTime();
 
-	  try {
-
-	    if(invoked){
-	      throw new RuntimeException("Can only invoke 'parse' once.");
-	    }
+		try {
+	    	if(invoked){
+	      		throw new RuntimeException("Can only invoke 'parse' once.");
+	    	}
 
+	    	invoked = true;
 
-	    invoked = true;
+	    	// Initialize.
+	    	this.inputURI = inputURI;
+	    	this.input = input;
 
-	    // Initialize.
-	    this.inputURI = inputURI;
-	    this.input = input;
-
-	    this.recoverer = recoverer;
+	    	this.recoverer = recoverer;
 			this.debugListener = debugListener;
 		
 			visualizer = ParseStateVisualizer.shouldVisualizeUri(inputURI) ? new ParseStateVisualizer("Parser") : null;
 
-	    // Initialzed the position store.
-	    positionStore.index(input);
+	    	// Initialzed the position store.
+	    	positionStore.index(input);
 
-	    todoLists = new DoubleStack[DEFAULT_TODOLIST_CAPACITY];
+	    	todoLists = new DoubleStack[DEFAULT_TODOLIST_CAPACITY];
 
-	    // Handle the initial expansion of the root node.
-	    AbstractStackNode<P> rootNode = startNode;
-	    rootNode.initEdges();
-	    stacksToExpand.push(rootNode);
-	    lookAheadChar = (input.length > 0) ? input[0] : 0;
+	    	// Handle the initial expansion of the root node.
+	    	AbstractStackNode<P> rootNode = startNode;
+	    	rootNode.initEdges();
+	    	stacksToExpand.push(rootNode);
+	    	lookAheadChar = (input.length > 0) ? input[0] : 0;
 
 			debugListener.shifting(location, input, positionStore);
 
-	    expand();
+	    	expand();
 
 			AbstractContainerNode<P> result = null;
-	    if(findFirstStacksToReduce()){
-	      boolean shiftedLevel = (location != 0);
+	    	if(findFirstStacksToReduce()){
+	      		boolean shiftedLevel = (location != 0);
 				while (true) {
-	        lookAheadChar = (location < input.length) ? input[location] : 0;
-	        if(shiftedLevel){ // Nullable fix for the first level.
-	          sharedNextNodes.clear();
-	          cachedEdgesForExpect.clear();
+	        		lookAheadChar = (location < input.length) ? input[location] : 0;
```

Comment:
these changes are more involved due to spaces getting into tabs, is that intended?